lib/statespace_basic.h

```diff
@@ -94,6 +94,40 @@ struct StateSpaceBasic final : public StateSpace<ParallelFor, FP> {
         Base::num_threads_, Base::size_, f, Op(), state);
   }
 
+  std::complex<double> InnerProduct(
+      const State& state1, const State& state2) const {
+    using Op = std::plus<std::complex<double>>;
+
+    auto f = [](unsigned n, unsigned m, uint64_t i, const State& state1,
+              const State& state2) -> std::complex<double> {
+      auto s1 = state1.get() + 2 * i;
+      auto s2 = state2.get() + 2 * i;
+
+      double re = s1[0] * s2[0] + s1[1] * s2[1];
+      double im = s1[0] * s2[1] - s1[1] * s2[0];
```

Comment:
This seems reversed to me - shouldn't the real component have a subtraction (i.e. `re1*re2 - im1*im2`) and the imaginary component be addition only?

Reply:
The fist state should be complex conjugate. So this is reversed.